### ISSUE TITLE
Make sure translation works in Installer Bundle

### ISSUE THREAD

**Issue by craigh** [zikula/core#2191]
Currently unknown whether translation works at all in CoreInstallerBundle. The method `\Zikula\Bundle\CoreInstallerBundle\Util\ControllerUtil::setupLang()` is currently unused and may be required for translation

A functional translation of the strings will be required to see if it works ;-)

**Comment by cmfcmf** [comment]
Make also sure the translation of the route works, depends on #2138.

**Comment by rallek** [comment]
Is an actual pot file available? I started the German translation for an older build

**Comment by craigh** [comment]
@Guite how do we regenerate the core pot file?

**Comment by craigh** [comment]
@Guite will you please regenerate the POT file and document how to do it for the rest of us?

**Comment by Guite** [comment]
Yes. Also I will check how much effort is creating a CI job doing the pot generation.

**Comment by Guite** [comment]
Pot files are regenerated in f98258fef0e0ded173df41a3435a2d3645d9da03

To do this:

1. Clone https://github.com/zikula-modules/Gettext/ into the modules/ folder
2. Open a console and navigate into `modules/Gettext/Helper`
3. Call `xtractcore.sh path/to/src/`

Going to challenge a CI job for this next.

**Comment by craigh** [comment]
maybe the core extract files could be moved right to the core repo in the /Tools dir. unless they also require other elements of the module.

**Comment by craigh** [comment]
the strings in the twig templates of the installer are not extracted: https://github.com/zikula-modules/Gettext/issues/8

**Comment by Guite** [comment]
It is not easily possible to separate only the core extract files.

**Comment by Guite** [comment]
The POT creation is now working as a proof of concept:
http://ci.zikula.org/job/Zikula_Core-1.4-POT/

**Comment by craigh** [comment]
@Guite - I've spent some time trying to figure out how to make the twig files work in our Gettext extractor and can't figure it out. it is WAY above my head. can you take a look?

**Comment by Guite** [comment]
No I am no regex guru.

**Comment by shefik** [comment]
@craigh Since you changed the milestone for this particular ticket to 1.4.1, I assume that means translations for the installer of Zikula 1.4.0 are not included in that release at all?

**Comment by craigh** [comment]
right. this is quite complicated actually.

**Comment by craigh** [comment]
@shefik @Kaik @rallek 

.pot files are now in place for the installer. It would be great if someone could create a .po file and test translation in the installer.

If it works, then we can close this ticket. If it doesn't then we will probably postpone (again) to 1.4.1. but if someone can test, that would be terrific!

**Comment by shefik** [comment]
I can test it.

**Comment by Kaik** [comment]
I will test hopefully by the weekend.

**Comment by craigh** [comment]
@shefik @Kaik ?

**Comment by craigh** [comment]
@shefik @Kaik @rallek can anyone test this in the next 7 days? I think it is functional and correct but would like some confirmation please. Else I will postpone this ticket again.

**Comment by rallek** [comment]
first I would need an actual translation ;-). I will give it a try with a translation of 1.3.5. Some strings should be availble already.

**Comment by rallek** [comment]
I tried without success!
Fresh download of ci build 113. uploaded the german translation into ``/app/Resources/locale/de/LC_MESSAGES`` (both po and mo files). and start web installer.

I do not get the chance to choose the language.

Next I opend the zikula.po in Poedit. Downloaded the actual pot file from ci server, updated the zikula.po in poedit. Now I want to translate the first page of installer. But I can not find the string ``Zikula Core is not installed!``! Is the installer not included in the translation?

**Comment by craigh** [comment]
you are editing the wrong file. :-)

these are the files to translate:

https://github.com/zikula/core/tree/1.4/src/lib/Zikula/Bundle/CoreInstallerBundle/Resources/translations

**Comment by craigh** [comment]
more info: https://github.com/zikula/core/blob/1.4/src/docs/en/dev/zikula_sf_translator.md

**Comment by rallek** [comment]
And where to place the translation? I am confused. How many translation files we do have? It is not included into pot file generation at ci server :-(

**Comment by craigh** [comment]
> In the future, the files must be placed into the new app/Resources/translations folder. In this case, copy all files to <filename>.<lang>.po (for example routes.template.po => routes.de.po) and translate the .po files as usual to generate the .mo files.

**Comment by rallek** [comment]
I tried this folder and the standard folder of zikula translation. Both do not work.

**Comment by rallek** [comment]
@Kaik will hopefully try next week. Translation makes trouble at all: #2642
